framework/src/problems/FEProblemBase.C

```diff
@@ -2294,6 +2310,35 @@ FEProblemBase::getFunction(const std::string & name, const THREAD_ID tid)
   return *ret;
 }
 
+bool
+FEProblemBase::hasConvergence(const std::string & name, const THREAD_ID tid)
+{
+  return _convergences.hasActiveObject(name, tid);
+}
+
+Convergence &
+FEProblemBase::getConvergence(const std::string & name, const THREAD_ID tid)
+{
+  // This thread lock is necessary since this method will create functions
+  // for all threads if one is missing.
+  // Threads::spin_mutex::scoped_lock lock(get_function_mutex);
+
+  if (!hasConvergence(name, tid))
+  {
+    // mooseError("Unable to find convergence criteria " + name);
+    std::string class_name = "ResidualConvergence";
+    InputParameters params = _factory.getValidParams(class_name);
+    setNonlinearConvergenceObject("default");
+    addConvergence(class_name, _nonlinear_convergence_name, params);
+  }
```

Comment:
I think we should create the default convergence somewhere else. Maybe put  this in a method called `addDefaultNonlinearConvergence()`, and then maybe in `FEProblemSolve`, do
```
if (isParamValid("nonlinear_convergence"))
{
_problem.setNonlinearConvergenceObject(getParam<ConvergenceName>("nonlinear_convergence"));
}
else
{
  _problem.setNonlinearConvergenceObject("default_convergence");
  addDefaultNonlinearConvergence();
}

Reply:
Thanks! Didn't like it like it this either, I mentioned it before, but never got to have a meeting on this. I try this.